emcc

```diff
@@ -673,6 +674,16 @@ try:
     elif newargs[i] == '--emrun':
       emrun = True
       newargs[i] = ''
+    elif newargs[i] == '--memprof':
+      tracing = True
+      memprof = True
+      newargs[i] = ''
+      newargs.append('-D__EMSCRIPTEN_TRACING__=1')
+      settings_changes.append('EMSCRIPTEN_TRACING=1')
+      js_libraries.append(shared.path_from_root('src', 'library_trace.js'))
+      js_libraries.append(shared.path_from_root('src', 'memprof_js_library.js'))
+      pre_js += open(shared.path_from_root('src', 'memprof_prejs.js')).read() + '\n'
```

Comment:
Could we make do without the use of --pre-js and --post-js lines here? The files are more or less empty anyways.